### PR TITLE
Initial commit for Manufacturer Component Sample

### PR DESCRIPTION
**Pull request by prateekcr**
- The code is essentially a configurable version of
protocol-samples/http-srver-di-sample. Certain properties are exposed
outside that can be passed as environment/system variables.
- Runnable manufacturer jar is created at mvn install phase that is
copied over to demo/manufacturer, along with the libs directory.
- Docker scripts have been added to demo/manufacturer.

Signed-off-by: Chandrakar, Prateek <prateek.chandrakar@intel.com>

### CODE REVIEW

**Review by Darshini-Parikh93 — APPROVED** [review]

File: service/component-samples/manufacturer/pom.xml [hunk 1]
```diff
@@ -0,0 +1,210 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <artifactId>protocol-samples</artifactId>
```
Comment (trbehera) [inline service/component-samples/manufacturer/pom.xml:8]: manufacturer?

File: service/component-samples/manufacturer/src/main/java/org/fido/iot/sample/ManufacturerContextListener.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+import org.fido.iot.storage.CertificateResolver;
+import org.fido.iot.storage.DiDbManager;
+import org.fido.iot.storage.DiDbStorage;
+
+/**
+ * Device Initialization servlet Context Listener.
+ */
+public class ManufacturerContextListener implements ServletContextListener {
+
+  private static final String mfgKeyPem = "-----BEGIN CERTIFICATE-----\n"
```
Comment (trbehera) [inline service/component-samples/manufacturer/src/main/java/org/fido/iot/sample/ManufacturerContextListener.java:33]: In a subsequent PR, make this configurable.

**Review by trbehera — CHANGES_REQUESTED** [review]

File: service/component-samples/manufacturer/pom.xml (hunk 1 above)
Comment (prateekcr) [reply service/component-samples/manufacturer/pom.xml]: Done.

File: service/component-samples/manufacturer/src/main/java/org/fido/iot/sample/ManufacturerContextListener.java (hunk 2 above)
Comment (prateekcr) [reply service/component-samples/manufacturer/src/main/java/org/fido/iot/sample/ManufacturerContextListener.java:33]: Agreed.

**Review by trbehera — APPROVED** [review]

**Review by Darshini-Parikh93 — APPROVED** [review]

**Pull request merged**